Makefile

```diff
@@ -71,26 +78,14 @@ lint-go:
 lint: ensure-submodule-initialized lint-cpp lint-go
 
 setup: ensure-submodule-initialized
-ifeq ("Darwin", "$(shell uname -s)")
-	cd viam-cartographer/scripts && ./setup_cartographer_macos.sh
-else
-	cd viam-cartographer/scripts && ./setup_cartographer_linux.sh
-endif
+	viam-cartographer/scripts/setup_cartographer.sh
 	@make artifact-pull
-	
-build: build-module
-ifneq ($(wildcard viam-cartographer/cartographer/build/.),)
-	cd viam-cartographer && ./scripts/build_viam_cartographer.sh 
-else 
-	cd viam-cartographer && ./scripts/build_cartographer.sh && ./scripts/build_viam_cartographer.sh
-endif
 
-build-debug: build-module
-ifneq ($(wildcard viam-cartographer/cartographer/build/.),)
-	cd viam-cartographer && ./scripts/build_viam_cartographer_debug.sh 
-else 
-	cd viam-cartographer && ./scripts/build_cartographer.sh && ./scripts/build_viam_cartographer_debug.sh
-endif
+build: ensure-submodule-initialized buf build-module
+	cd viam-cartographer && cmake -Bbuild -G Ninja ${EXTRA_CMAKE_FLAGS} && cmake --build build
```

Comment:
we dont have to run the `ninja` command anymore?

Reply:
Yeah, in theory you can just change -Bbuild to -Bbuild-\`uname -m\`  (or whatever name scheme you want, and it'll configure the output there. Then you can actually compile with the same thing just changing build to build-\`uname -m\` in the final command as well.

More realistically, you'd just want to make them variables in Make, then you can set the BUILD_DIR variable once and use everywhere. Do note that some of the cartographer tests in go are written using cgo and depending on explicit paths (which is why I had to change one) so there's not as easy of a way to handle that.

Likewise, it's not just the cmake part that would need to be per-architecture, as the golang stuff would be as well, so it's likely a bit of a more complex change you really want, to completely split out architectures if you really want to be able to build both in the same code space (without just doing "make clean" instead.)